src/renderer/viz/expressions/ramp.js

```diff
@@ -97,28 +123,63 @@ export default class Ramp extends BaseExpression {
         this.palette._setUID(idGenerator);
     }
 
-    eval(o) {
-        if (this.palette.type != 'number-array') {
-            super.eval(o);
+    eval(feature) {
+        const pixelValues = this._computeTextureIfNeeded();
+        const input = this.input.eval(feature);
+
+        if (input !== -1) {
+            const numValues = pixelValues.length - 1;
+            const m = (input - this.minKey) / (this.maxKey - this.minKey);
+            
+            const color = this.type === rampTypes.NUMBER
+                ? this._getValue(pixelValues, numValues, m)
+                : this._getColorValue(pixelValues, m);
+            
+            this._texCategories = null;
+            this._GLtexCategories = null;
+
+            return color;
         }
-        this._computeTextureIfNeeded();
-        const input = this.input.eval(o);
-        const m = (input - this.minKey) / (this.maxKey - this.minKey);
-        const len = this.pixel.length - 1;
-        const lowIndex = clamp(Math.floor(len * m), 0, len);
-        const highIndex = clamp(Math.ceil(len * m), 0, len);
-        const low = this.pixel[lowIndex];
-        const high = this.pixel[highIndex];
-        const fract = len * m - Math.floor(len * m);
-        return fract * high + (1 - fract) * low;
+
+        return null;
+    }
+
+    _getValue(pixelValues, numValues, m) {
+        const lowIndex = clamp(Math.floor(numValues * m), 0, numValues);
+        const highIndex = clamp(Math.ceil(numValues * m), 0, numValues);
+        const fract = numValues * m - Math.floor(numValues * m);
+        const low = pixelValues[lowIndex];
+        const high = pixelValues[highIndex];
+
+        return Math.round(fract * high + (1 - fract) * low);
+    }
+    
+    _getColorValue(pixelValues, m) {
+        const index = Math.round(m * MAX_BYTE_VALUE);
```

Comment:
I do not think so because otherwise there will be indexes out of the array. Actually, the bad thing here is the naming... probably it should be `COLOR_ARRAY_LENGTH - 1` (which in this case is the same as MAX_BYTE_VALUE, but that is a coincidence in this case).